src/staking/delegation.service.ts

```diff
@@ -0,0 +1,109 @@
+import { HttpService } from '@nestjs/axios';
+import {
+  ForbiddenException,
+  Injectable,
+  InternalServerErrorException,
+  Logger,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { BigNumber } from 'ethers';
+import { catchError, firstValueFrom, map } from 'rxjs';
+
+@Injectable()
+export class DelegationService {
+  private readonly logger = new Logger(DelegationService.name);
+  constructor(
+    private configService: ConfigService,
+    private readonly httpService: HttpService,
+  ) {}
+
+  async totalDelegated() {
+    const delegationApi = `${this.configService.get<string>(
+      'delegationAPIRoot',
+    )}/validators?show_fully_delegated=true&show_my_validators=false`;
+    let delegationData;
+    try {
+      const realTimeDelegationData = await firstValueFrom(
+        this.httpService
+          .get(delegationApi)
+          .pipe(
+            catchError((error) => {
+              this.logger.error(error);
+              throw new ForbiddenException('API not available');
+            }),
+          )
+          .pipe(
+            map((res) => {
+              return res.data;
+            }),
+          ),
+      );
+      delegationData = realTimeDelegationData;
+    } catch (error) {
+      this.logger.error(error);
+      throw new InternalServerErrorException(
```

Comment:
Remove this InternalServerErrorException otherwise it will make the entire endpoint fail. Just leave the logger.

Reply:
This is exactly what we want as in this scenario we want to return 500 as a code not 200